### PR TITLE
feat(onboarding-dashboard-templates): get the iframe loading

### PR DESCRIPTION
**Pull request by raquelmsmith**
## Problem

https://github.com/PostHog/posthog/issues/23065

We need to show an iframe with the customer's site so they can select their actions

## Changes

- [x] Detect hosts where posthog-js is installed
- [x] automatically authorize those domains when selected
- [x] Load host site in iframe
    - [x] Abstract toolbar-iframe-specific code form heatmaps

Arc 2024-08-29 14 20 27

Arc 2024-08-29 14 20 39

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Does this work well for both Cloud and self-hosted?

Should be fine for now

## How did you test this code?

👀

### CODE REVIEW

**Comment by raquelmsmith** [comment]
@pauldambra I'm going to reuse the heatmap iframe code because it has a lot of plumbing I am going to need for this project. I abstracted it out into its own logic and component, and left the heatmap logic only stuff to do with heatmaps specifically (like loading possible URLs, or toggling the filter pane). Things seem to work locally but lmk if something seems off.

File: frontend/src/lib/components/IframedToolbarBrowser/IframedToolbarBrowser.tsx
Comment (raquelmsmith) [inline frontend/src/lib/components/IframedToolbarBrowser/IframedToolbarBrowser.tsx:1]: This is all just moved over from heatmaps-specific implementation

**Review by zlwaterfield — COMMENTED** [review]
Overall looks good. I don't have any feedback on any specific code. 

Few comments from testing locally:
- When I click blank dashboard I'm taken to http://localhost:8000/project/8/products
- When I click the product analytics dashboard I'm take to the plans step

I couldn't figure out why but this doesn't see to be working on my end. I have the test flag set. I wanted to test the iframe but couldn't get to it.

**Comment by raquelmsmith** [comment]
You have to add a dashboard template with variables in it to get to the iframe screen. I still want to redo the design for that bit entirely, so when I do that I will 1) hide `blank dashboard` and 2) make it clear somehow that your dashboard (without variables) was created before moving to the plans screen.

**Comment by raquelmsmith** [comment]
To add dashboard template:

Create a new dashboard. Go to dashboard menu/options button -> Save as Template. Paste the following in there and save.

```
{
    "id": "01872f58-c2e1-0000-16a0-27f39f896da1",
    "template_name": "AARRR – Pirate Metrics",
    "dashboard_description": "The AARRR framework, also known as \"pirate metrics\" or the AARRR funnel, is a classic framework for understanding customer behavior. It's useful for marketers, product managers and growth hackers at startups.",
    "dashboard_filters": {},
    "tags": [],
    "tiles": [
        {
            "body": "### Acquisition Metrics\n\nAcquisition metrics vary depending on product type.\n\nOverall unique users is an important metric for an e-commerce website, but user signups is better for B2B products. \n\nExamples of acquisition metrics include: \n\n1. Account signups\n2. App downloads\n3. Website visitors\n4. Signup conversion rate\n\nEdit these insights to use the acquisition metrics you care about most.\n\nOur guide to pirate metrics:\nhttps://posthog.com/blog/aarrr-pirate-funnel",
            "type": "TEXT",
            "color": null,
            "layouts": {
                "sm": {
                    "h": 6,
                    "i": "257225",
                    "w": 3,
                    "x": 0,
                    "y": 0,
                    "minH": 2,
                    "minW": 3,
                    "moved": false,
                    "static": false
                },
                "xs": {
                    "h": 6,
                    "i": "257225",
                    "w": 1,
                    "x": 0,
                    "y": 15,
                    "minH": 2,
                    "minW": 1
                }
            }
        },
        {
            "name": "[Referral] Invite to Signup Conversion Rate",
            "type": "INSIGHT",
            "color": null,
            "query": null,
            "filters": {
                "events": [
                    "{INVITED}",
                    "{SIGNED_UP}"
                ],
                "actions": [],
                "insight": "FUNNELS",
                "interval": "week",
                "date_from": "-90d",
                "exclusions": [],
                "new_entity": [],
                "properties": [],
                "funnel_viz_type": "trends",
                "filter_test_accounts": true
            },
            "layouts": {
                "sm": {
                    "h": 5,
                    "i": "257468",
                    "w": 4,
                    "x": 8,
                    "y": 23,
                    "minH": 5,
                    "minW": 3,
                    "moved": false,
                    "static": false
                },
                "xs": {
                    "h": 5,
                    "i": "257468",
                    "w": 1,
                    "x": 0,
                    "y": 64,
                    "minH": 5,
                    "minW": 1
                }
            },
            "description": "Historical trend of user signups via invites"
        },
        {
            "name": "[Referrals] Invites Sent",
            "type": "INSIGHT",
            "color": null,
            "query": null,
            "filters": {
                "events": [
                    "{INVITED}"
                ],
                "actions": [],
                "date_to": null,
                "display": "ActionsBar",
                "insight": "TRENDS",
                "interval": "week",
                "date_from": "-90d",
                "new_entity": [],
                "properties": [],
                "filter_test_accounts": true
            },
            "layouts": {
                "sm": {
                    "h": 5,
                    "i": "257465",
                    "w": 5,
                    "x": 3,
                    "y": 23,
                    "minH": 5,
                    "minW": 3,
                    "moved": false,
                    "static": false
                },
                "xs": {
                    "h": 5,
                    "i": "257465",
                    "w": 1,
                    "x": 0,
                    "y": 59,
                    "minH": 5,
                    "minW": 1
                }
            },
            "description": "Number of invites sent by active users"
        },
        {
            "body": "### Revenue Metrics\n\nWhat you track here will depend on your business.\n\nMonthly recurring revenue (MRR) is good for a B2B SaaS products. B2C companies businesses may prefer average revenue per user (ARPU), while e-commerce platforms prefer Average Basket Value (ABV).\n\nYou can also track revenue retention – i.e. whether customers continue to subscribe after paying their first invoice.\n\nEdit these insights to use the revenue metrics you care about most.",
            "type": "TEXT",
            "color": null,
            "layouts": {
                "sm": {
                    "h": 5,
                    "i": "257455",
                    "w": 3,
                    "x": 0,
                    "y": 18,
                    "minH": 2,
                    "minW": 3,
                    "moved": false,
                    "static": false
                },
                "xs": {
                    "h": 6,
                    "i": "257455",
                    "w": 1,
                    "x": 0,
                    "y": 15,
                    "minH": 2,
                    "minW": 1
                }
            }
        },
        {
            "name": "[Revenue] Free to Paid User Conversion",
            "type": "INSIGHT",
            "color": null,
            "query": null,
            "filters": {
                "events": [
                    "{SUBSCRIBED}",
                    "{SUB_PAID}"
                ],
                "layout": "horizontal",
                "actions": [],
                "insight": "FUNNELS",
                "date_from": "-90d",
                "exclusions": [],
                "new_entity": [],
                "properties": [],
                "funnel_viz_type": "steps",
                "filter_test_accounts": true,
                "funnel_window_interval": 6,
                "funnel_window_interval_unit": "week"
            },
            "layouts": {
                "sm": {
                    "h": 5,
                    "i": "257442",
                    "w": 5,
                    "x": 3,
                    "y": 18,
                    "minH": 5,
                    "minW": 3,
                    "moved": false,
                    "static": false
                },
                "xs": {
                    "h": 5,
                    "i": "257442",
                    "w": 1,
                    "x": 0,
                    "y": 49,
                    "minH": 5,
                    "minW": 1
                }
            },
            "description": "Tracks users who converted from free users to paid users with a 6-week conversion window"
        },
        {
            "name": "Weekly User Retention",
            "type": "INSIGHT",
            "color": null,
            "query": null,
            "filters": {
                "period": "Week",
                "insight": "RETENTION",
                "date_from": "-7d",
                "properties": [],
                "target_entity": "{LOGGED_ID}",
                "retention_type": "retention_recurring",
                "total_intervals": 9,
                "returning_entity": "{LOGGED_ID}",
                "retention_reference": "total",
                "filter_test_accounts": true
            },
            "layouts": {
                "sm": {
                    "h": 7,
                    "i": "257336",
                    "w": 9,
                    "x": 3,
                    "y": 11,
                    "minH": 5,
                    "minW": 3,
                    "moved": false,
                    "static": false
                },
                "xs": {
                    "h": 8,
                    "i": "257336",
                    "w": 1,
                    "x": 0,
                    "y": 41,
                    "minH": 5,
                    "minW": 1
                }
            },
            "description": "Users who logged in and then logged in again within an 8-week period"
        },
        {
            "name": "Activation Conversion (Trend)",
            "type": "INSIGHT",
            "color": null,
            "query": null,
            "filters": {
                "events": [
                    "{SIGNED_UP}",
                    "{LOGGED_ID}"
                ],
                "layout": "horizontal",
                "actions": [],
                "insight": "FUNNELS",
                "interval": "day",
                "date_from": "-90d",
                "exclusions": [],
                "new_entity": [],
                "properties": [],
                "funnel_viz_type": "trends",
                "filter_test_accounts": true
            },
            "layouts": {
                "sm": {
                    "h": 5,
                    "i": "257306",
                    "w": 4,
                    "x": 8,
                    "y": 6,
                    "minH": 5,
                    "minW": 3,
                    "moved": false,
                    "static": false
                },
                "xs": {
                    "h": 5,
                    "i": "257306",
                    "w": 1,
                    "x": 0,
                    "y": 36,
                    "minH": 5,
                    "minW": 1
                }
            },
            "description": "Historical trend of activation conversion"
        },
        {
            "body": "### Retention Metrics\n\nFor most products, measuring retention means defining a usage interval – i.e. how often you expect your retention event to occur. Do you expect users to come back every day, week or month.\n\nYou can also track unbounded retention, where you track if/how often users return without a specific interval.\n\n1. Complete a specific action in their first week\n2. Engage with your core feature\n3. Complete the onboarding process. \n\nEdit this insight to use the retention metric you care about most.",
            "type": "TEXT",
            "color": null,
            "layouts": {
                "sm": {
                    "h": 7,
                    "i": "257430",
                    "w": 3,
                    "x": 0,
                    "y": 11,
                    "minH": 2,
                    "minW": 3,
                    "moved": false,
                    "static": false
                },
                "xs": {
                    "h": 6,
                    "i": "257430",
                    "w": 1,
                    "x": 0,
                    "y": 15,
                    "minH": 2,
                    "minW": 1
                }
            }
        },
        {
            "body": "### Referral Metrics\n\nReferral metrics are important for products or services that have a growth loop – i.e. when the product experience is improved through multiple users, or when friends are using it together.\n\nEdit these insights to track the referral metric you care about most.",
            "type": "TEXT",
            "color": null,
            "layouts": {
                "sm": {
                    "h": 5,
                    "i": "257200",
                    "w": 3,
                    "x": 0,
                    "y": 23,
                    "minH": 2,
                    "minW": 3,
                    "moved": false,
                    "static": false
                },
                "xs": {
                    "h": 6,
                    "i": "257200",
                    "w": 1,
                    "x": 0,
                    "y": 15,
                    "minH": 2,
                    "minW": 1
                }
            }
        },
        {
            "name": "[Revenue] Total Paying Customers",
            "type": "INSIGHT",
            "color": null,
            "query": null,
            "filters": {
                "events": [
                    "{SUBSCRIBED}"
                ],
                "actions": [],
                "date_to": null,
                "display": "ActionsLineGraphCumulative",
                "insight": "TRENDS",
                "interval": "week",
                "date_from": "-90d",
                "new_entity": [],
                "properties": [],
                "filter_test_accounts": true
            },
            "layouts": {
                "sm": {
                    "h": 5,
                    "i": "257447",
                    "w": 4,
                    "x": 8,
                    "y": 18,
                    "minH": 5,
                    "minW": 3,
                    "moved": false,
                    "static": false
                },
                "xs": {
                    "h": 5,
                    "i": "257447",
                    "w": 1,
                    "x": 0,
                    "y": 54,
                    "minH": 5,
                    "minW": 1
                }
            },
            "description": "Cumulative growth of paying customers"
        },
        {
            "body": "### Activation Metrics\n\nTracking activation helps you isolate onboarding issues from long term retention issues. \n\nDon't be afraid to go deep here. Good activation metrics capture the \"aha\" moment a user have when they get value from your product.\n\nExamples of activation metrics include new users who: \n\n1. Complete a specific action in their first week\n2. Engage with your core feature\n3. Complete the onboarding process. \n\nEdit these insights to use the activation metrics you care about most.",
            "type": "TEXT",
            "color": null,
            "layouts": {
                "sm": {
                    "h": 5,
                    "i": "257320",
                    "w": 3,
                    "x": 0,
                    "y": 6,
                    "minH": 2,
                    "minW": 3,
                    "moved": false,
                    "static": false
                },
                "xs": {
                    "h": 6,
                    "i": "257320",
                    "w": 1,
                    "x": 0,
                    "y": 15,
                    "minH": 2,
                    "minW": 1
                }
            }
        },
        {
            "name": "Acquisition Conversion (Trend)",
            "type": "INSIGHT",
            "color": "black",
            "query": null,
            "filters": {
                "events": [
                    {
                        "id": "$pageview",
                        "name": "$pageview",
                        "type": "events",
                        "order": 0,
                        "custom_name": "Website Visitors"
                    },
                    {
                        "id": "$pageview",
                        "name": "$pageview",
                        "type": "events",
                        "order": 1,
                        "properties": [
                            {
                                "key": "$current_url",
                                "type": "event",
                                "value": "signup",
                                "operator": "icontains"
                            }
                        ],
                        "custom_name": "Signup Page Visitors"
                    },
                    "{SIGNED_UP}"
                ],
                "layout": "vertical",
                "actions": [],
                "insight": "FUNNELS",
                "interval": "week",
                "date_from": "-90d",
                "exclusions": [],
                "new_entity": [],
                "properties": [],
                "funnel_viz_type": "trends",
                "filter_test_accounts": true
            },
            "layouts": {
                "sm": {
                    "h": 6,
                    "i": "257206",
                    "w": 4,
                    "x": 8,
                    "y": 0,
                    "minH": 5,
                    "minW": 3,
                    "moved": false,
                    "static": false
                },
                "xs": {
                    "h": 5,
                    "i": "257206",
                    "w": 1,
                    "x": 0,
                    "y": 26,
                    "minH": 5,
                    "minW": 1
                }
            },
            "description": "Historical trend of your acquisition funnel"
        },
        {
            "name": "Acquisition Conversion",
            "type": "INSIGHT",
            "color": "black",
            "query": null,
            "filters": {
                "events": [
                    {
                        "id": "$pageview",
                        "name": "$pageview",
                        "type": "events",
                        "order": 0,
                        "custom_name": "Website Visitors"
                    },
                    {
                        "id": "$pageview",
                        "name": "$pageview",
                        "type": "events",
                        "order": 1,
                        "properties": [
                            {
                                "key": "$current_url",
                                "type": "event",
                                "value": "signup",
                                "operator": "icontains"
                            }
                        ],
                        "custom_name": "Signup Page Visitors"
                    },
                    "{SIGNED_UP}"
                ],
                "layout": "horizontal",
                "actions": [],
                "insight": "FUNNELS",
                "date_from": "-90d",
                "exclusions": [],
                "new_entity": [],
                "properties": [],
                "funnel_viz_type": "steps",
                "filter_test_accounts": true,
                "funnel_window_interval": 6,
                "funnel_window_interval_unit": "week"
            },
            "layouts": {
                "sm": {
                    "h": 6,
                    "i": "257201",
                    "w": 5,
                    "x": 3,
                    "y": 0,
                    "minH": 5,
                    "minW": 3,
                    "moved": false,
                    "static": false
                },
                "xs": {
                    "h": 5,
                    "i": "257201",
                    "w": 1,
                    "x": 0,
                    "y": 21,
                    "minH": 5,
                    "minW": 1
                }
            },
            "description": "Conversions from website visits, to signup page, and user signup"
        },
        {
            "name": "Activation Conversion",
            "type": "INSIGHT",
            "color": null,
            "query": null,
            "filters": {
                "events": [
                    "{SIGNED_UP}",
                    "{LOGGED_ID}"
                ],
                "layout": "horizontal",
                "actions": [],
                "insight": "FUNNELS",
                "date_from": "-90d",
                "exclusions": [],
                "new_entity": [],
                "properties": [],
                "funnel_viz_type": "steps",
                "filter_test_accounts": true,
                "funnel_window_interval": 1,
                "funnel_window_interval_unit": "week"
            },
            "layouts": {
                "sm": {
                    "h": 5,
                    "i": "257292",
                    "w": 5,
                    "x": 3,
                    "y": 6,
                    "minH": 5,
                    "minW": 3,
                    "moved": false,
                    "static": false
                },
                "xs": {
                    "h": 5,
                    "i": "257292",
                    "w": 1,
                    "x": 0,
                    "y": 31,
                    "minH": 5,
                    "minW": 1
                }
            },
            "description": "Tracking conversion from signup to users logging in within one week"
        }
    ],
    "variables": [
        {
            "id": "INVITED",
            "name": "Invited teammate",
            "type": "event",
            "default": {},
            "required": true,
            "description": "Inviting a teammmate to join the product"
        },
        {
            "id": "SIGNED_UP",
            "name": "User signed up",
            "type": "event",
            "default": {},
            "required": true,
            "description": "User signing up"
        },
        {
            "id": "SUBSCRIBED",
            "name": "User subscribed",
            "type": "event",
            "default": {},
            "required": true,
            "description": "A subscription activated for a user"
        },
        {
            "id": "SUB_PAID",
            "name": "User subscription paid",
            "type": "event",
            "default": {},
            "required": true,
            "description": "A subscription paid by a user"
        },
        {
            "id": "LOGGED_ID",
            "name": "User logged in",
            "type": "event",
            "default": {},
            "required": true,
            "description": "A user login"
        }
    ],
    "deleted": null,
    "created_at": "2023-03-29T21:49:18.946507Z",
    "created_by": null,
    "image_url": "https://posthog.com/template_art/aarrr-dash.png",
    "team_id": 2,
    "scope": "global"
}
```

**Comment by zlwaterfield** [comment]
Awesome thank you!

**Review by zlwaterfield — APPROVED** [review]

**Comment by pauldambra** [comment]
sorry to miss this... pretty exciting feature 🙌 !

**Pull request merged**